cli/src/description_util.rs

```diff
@@ -40,6 +45,110 @@ JJ: Lines starting with "JJ: " (like this one) will be removed.
     Ok(text_util::complete_newline(description.trim_matches('\n')))
 }
 
+#[derive(Debug)]
+pub struct EditMultipleDescriptionsResult {
+    /// The parsed, formatted descriptions.
+    pub descriptions: HashMap<CommitId, String>,
+    /// Commit IDs that were found multiple times while parsing the edited
+    /// messages.
+    pub duplicates: Vec<String>,
+    /// Commit IDs that were found while parsing the edited messages, but which
+    /// were not originally being edited.
+    pub unexpected: Vec<String>,
+}
+
+/// Edits the descriptions of the given commits in a single editor session.
+pub fn edit_multiple_descriptions(
+    ui: &Ui,
+    settings: &UserSettings,
+    workspace_command: &WorkspaceCommandHelper,
+    repo: &ReadonlyRepo,
+    commits: &[Commit],
+) -> Result<EditMultipleDescriptionsResult, CommandError> {
+    let id_prefix_context = workspace_command.id_prefix_context()?;
+    let mut commits_map = HashMap::new();
+    let mut output_chunks = Vec::new();
+
+    for commit in commits.iter().rev() {
+        let commit_id_prefix = generate_commit_id_prefix(id_prefix_context, repo, commit);
```

Comment:
Isn't it better to use `short_commit_has()` or full hex? I don't think the shortest id helps here. "JJ: describe 8" is rather cryptic.

Reply:
I was wondering if we were ever worried about conflicting commit IDs, which was why I used the ID prefix. The available alternatives are `short_commit_hash()`, full ID, or use the prefix with a minimum length of eg. 12 (for consistency with `short_commit_hash`). Any opinions?

Reply:
I think `short_commit_hash()` is pretty good here. It's long enough to recognize as a hex string, and is unlikely to become ambiguous within tens of commits to be edited in bulk.